recipes/cairo/meson/conanfile.py

```diff
@@ -121,7 +122,19 @@ def validate(self):
 
     @property
     def _is_msvc(self):
-        return self.settings.compiler == "Visual Studio"
+        return str(self.settings.compiler) in ["msvc", "Visual Studio"]
+
+    @contextlib.contextmanager
+    def _build_context(self):
+        if self._is_msvc:
+            env_build = VisualStudioBuildEnvironment(self)
+            if not self.options.shared:
+                env_build.flags.append("-DCAIRO_WIN32_STATIC_BUILD")
+                env_build.cxx_flags.append("-DCAIRO_WIN32_STATIC_BUILD")
+            with tools.environment_append(env_build.vars):
+                yield
+        else:
+            yield
```

Comment:
Why is it required to get vars from `VisualStudioBuildEnvironment` then inject them for Meson? Why non-windows builds don't need to inject any include path of dependencies with something similar?

Reply:
I've checked some of them, and they were already using `VisualStudioBuildEnvironment`